### ISSUE TITLE
FAIL test: TestConfigLoadConfiguration 

### ISSUE THREAD

**Issue by zhsj** [kata-containers/kata-containers#1443]
```
$ go test -mod=vendor -run '^TestConfigLoadConfiguration$' ./pkg/katautils/
INFO: creating test directory
INFO: test directory is /tmp/kata-794693374
INFO: creating OCI bundle in /tmp/kata-794693374/bundle for tests to use
INFO: checking for container engine: docker
--- FAIL: TestConfigLoadConfiguration (0.00s)
    config_test.go:281: Expected
        {HypervisorType:qemu HypervisorConfig:{NumVCPUs:1 DefaultMaxVCPUs:4 MemorySize:2048 DefaultBridges:1 Msize9p:8192 MemSlots:10 MemOffset:0 VirtioFSCacheSize:0 KernelParams:[{Key:systemd.unit Value:kata-containers.target} {Key:systemd.mask Value:systemd-networkd.service} {Key:systemd.mask Value:systemd-networkd.socket} {Key:scsi_mod.scan Value:none} {Key:foo Value:bar} {Key:xyz Value:}] HypervisorParams:[] KernelPath:/tmp/kata-794693374/load-config-039136325/test/kernel ImagePath:/tmp/kata-794693374/load-config-039136325/test/image InitrdPath: FirmwarePath: MachineAccelerators: CPUFeatures: HypervisorPath:/tmp/kata-794693374/load-config-039136325/test/hypervisor HypervisorPathList:[] HypervisorCtlPathList:[] HypervisorCtlPath: JailerPath: JailerPathList:[] BlockDeviceDriver:virtio-scsi HypervisorMachineType:machineType MemoryPath: DevicesStatePath: EntropySource:/dev/urandom SharedFS:virtio-9p VirtioFSDaemon:/tmp/kata-794693374/load-config-039136325/test/virtiofsd VirtioFSDaemonList:[] VirtioFSCache:none VirtioFSExtraArgs:[] FileBackedMemRootDir: FileBackedMemRootList:[] PFlash:[] customAssets:map[] BlockDeviceCacheSet:false BlockDeviceCacheDirect:false BlockDeviceCacheNoflush:false DisableBlockDeviceUse:true EnableIOThreads:true Debug:false MemPrealloc:false HugePages:false VirtioMem:false IOMMU:false IOMMUPlatform:false Realtime:false Mlock:true DisableNestingChecks:false DisableImageNvdimm:false HotplugVFIOOnRootBus:true PCIeRootPort:2 BootToBeTemplate:false BootFromTemplate:false DisableVhostNet:false EnableVhostUserStore:false VhostUserStorePath:/var/run/kata-containers/vhost-user/ VhostUserStorePathList:[] GuestHookPath: VMid: SELinuxProcessLabel: RxRateLimiterMaxRate:0 TxRateLimiterMaxRate:0 SGXEPCSize:0 EnableAnnotations:[] GuestMemoryDumpPath: GuestMemoryDumpPaging:false} NetmonConfig:{Path:/tmp/kata-794693374/load-config-039136325/test/netmon Debug:false Enable:false} AgentConfig:{LongLiveConn:true Debug:false Trace:false EnableDebugConsole:false ContainerPipeSize:0 TraceMode: TraceType: KernelModules:[]} Console: InterNetworkModel:0 FactoryConfig:{Template:false TemplatePath:/run/vc/vm/template VMCacheNumber:0 VMCacheEndpoint:/var/run/kata-containers/cache.sock} Debug:false Trace:false JaegerEndpoint:localhost JaegerUser:jaeger_user1 JaegerPassword:jaeger_password1 DisableGuestSeccomp:false DisableNewNetNs:false SandboxCgroupOnly:false Experimental:[] EnablePprof:true}
        Got
        {HypervisorType:qemu HypervisorConfig:{NumVCPUs:1 DefaultMaxVCPUs:4 MemorySize:2048 DefaultBridges:1 Msize9p:8192 MemSlots:10 MemOffset:0 VirtioFSCacheSize:0 KernelParams:[{Key:systemd.unit Value:kata-containers.target} {Key:systemd.mask Value:systemd-networkd.service} {Key:systemd.mask Value:systemd-networkd.socket} {Key:scsi_mod.scan Value:none} {Key:foo Value:bar} {Key:xyz Value:}] HypervisorParams:[] KernelPath:/tmp/kata-794693374/load-config-039136325/test/kernel ImagePath:/tmp/kata-794693374/load-config-039136325/test/image InitrdPath: FirmwarePath: MachineAccelerators: CPUFeatures: HypervisorPath:/tmp/kata-794693374/load-config-039136325/test/hypervisor HypervisorPathList:[] HypervisorCtlPathList:[] HypervisorCtlPath: JailerPath: JailerPathList:[] BlockDeviceDriver:virtio-scsi HypervisorMachineType:machineType MemoryPath: DevicesStatePath: EntropySource:/dev/urandom SharedFS:virtio-9p VirtioFSDaemon:/tmp/kata-794693374/load-config-039136325/test/virtiofsd VirtioFSDaemonList:[] VirtioFSCache:none VirtioFSExtraArgs:[] FileBackedMemRootDir: FileBackedMemRootList:[] PFlash:[] customAssets:map[] BlockDeviceCacheSet:false BlockDeviceCacheDirect:false BlockDeviceCacheNoflush:false DisableBlockDeviceUse:true EnableIOThreads:true Debug:false MemPrealloc:false HugePages:false VirtioMem:false IOMMU:false IOMMUPlatform:false Realtime:false Mlock:true DisableNestingChecks:false DisableImageNvdimm:false HotplugVFIOOnRootBus:true PCIeRootPort:2 BootToBeTemplate:false BootFromTemplate:false DisableVhostNet:false EnableVhostUserStore:false VhostUserStorePath:/var/run/kata-containers/vhost-user/ VhostUserStorePathList:[] GuestHookPath: VMid: SELinuxProcessLabel: RxRateLimiterMaxRate:0 TxRateLimiterMaxRate:0 SGXEPCSize:0 EnableAnnotations:[] GuestMemoryDumpPath: GuestMemoryDumpPaging:false} NetmonConfig:{Path:/tmp/kata-794693374/load-config-039136325/test/netmon Debug:false Enable:false} AgentConfig:{LongLiveConn:true Debug:false Trace:false EnableDebugConsole:false ContainerPipeSize:0 TraceMode: TraceType: KernelModules:[]} Console: InterNetworkModel:0 FactoryConfig:{Template:false TemplatePath:/run/vc/vm/template VMCacheNumber:0 VMCacheEndpoint:/var/run/kata-containers/cache.sock} Debug:false Trace:false JaegerEndpoint:localhost JaegerUser:jaeger_user1 JaegerPassword:jaeger_password1 DisableGuestSeccomp:false DisableNewNetNs:false SandboxCgroupOnly:false Experimental:[] EnablePprof:true}
FAIL
FAIL    github.com/kata-containers/kata-containers/src/runtime/pkg/katautils    0.283s
FAIL
```

The log is not readable. If I change the `reflect.DeepEqual` to `assert.Equal`, the error is much readable.

```diff
diff --git a/src/runtime/pkg/katautils/config_test.go b/src/runtime/pkg/katautils/config_test.go
index 6d454a73..4cf85fb3 100644
--- a/src/runtime/pkg/katautils/config_test.go
+++ b/src/runtime/pkg/katautils/config_test.go
@@ -276,6 +276,7 @@ func testLoadConfiguration(t *testing.T, dir string,
                                }
 
                                assert.Equal(t, defaultRuntimeConfiguration, resolvedConfigPath)
+                               assert.Equal(t, testConfig.RuntimeConfig, config)
                                result := reflect.DeepEqual(config, testConfig.RuntimeConfig)
                                if !result {
                                        t.Fatalf("Expected\n%+v\nGot\n%+v", config, testConfig.RuntimeConfig)

```

```
$ go test -mod=vendor -run '^TestConfigLoadConfiguration$' ./pkg/katautils/
INFO: creating test directory
INFO: test directory is /tmp/kata-597178406
INFO: creating OCI bundle in /tmp/kata-597178406/bundle for tests to use
INFO: checking for container engine: docker
--- FAIL: TestConfigLoadConfiguration (0.00s)
    config_test.go:279: 
                Error Trace:    config_test.go:279
                                                        config_test.go:298
                Error:          Not equal: 
                                expected: oci.RuntimeConfig{HypervisorType:"qemu", HypervisorConfig:virtcontainers.HypervisorConfig{NumVCPUs:0x1, DefaultMaxVCPUs:0x4, MemorySize:0x800, DefaultBridges:0x1, Msize9p:0x2000, MemSlots:0xa, MemOffset:0x0, VirtioFSCacheSize:0x0, KernelParams:[]virtcontainers.Param{virtcontainers.Param{Key:"systemd.unit", Value:"kata-containers.target"}, virtcontainers.Param{Key:"systemd.mask", Value:"systemd-networkd.service"}, virtcontainers.Param{Key:"systemd.mask", Value:"systemd-networkd.socket"}, virtcontainers.Param{Key:"scsi_mod.scan", Value:"none"}, virtcontainers.Param{Key:"foo", Value:"bar"}, virtcontainers.Param{Key:"xyz", Value:""}}, HypervisorParams:[]virtcontainers.Param(nil), KernelPath:"/tmp/kata-597178406/load-config-133133645/test/kernel", ImagePath:"/tmp/kata-597178406/load-config-133133645/test/image", InitrdPath:"", FirmwarePath:"", MachineAccelerators:"", CPUFeatures:"", HypervisorPath:"/tmp/kata-597178406/load-config-133133645/test/hypervisor", HypervisorPathList:[]string(nil), HypervisorCtlPathList:[]string(nil), HypervisorCtlPath:"", JailerPath:"", JailerPathList:[]string(nil), BlockDeviceDriver:"virtio-scsi", HypervisorMachineType:"machineType", MemoryPath:"", DevicesStatePath:"", EntropySource:"/dev/urandom", SharedFS:"virtio-9p", VirtioFSDaemon:"/tmp/kata-597178406/load-config-133133645/test/virtiofsd", VirtioFSDaemonList:[]string(nil), VirtioFSCache:"none", VirtioFSExtraArgs:[]string(nil), FileBackedMemRootDir:"", FileBackedMemRootList:[]string(nil), PFlash:[]string(nil), customAssets:map[types.AssetType]*types.Asset(nil), BlockDeviceCacheSet:false, BlockDeviceCacheDirect:false, BlockDeviceCacheNoflush:false, DisableBlockDeviceUse:true, EnableIOThreads:true, Debug:false, MemPrealloc:false, HugePages:false, VirtioMem:false, IOMMU:false, IOMMUPlatform:false, Realtime:false, Mlock:true, DisableNestingChecks:false, DisableImageNvdimm:false, HotplugVFIOOnRootBus:true, PCIeRootPort:0x2, BootToBeTemplate:false, BootFromTemplate:false, DisableVhostNet:false, EnableVhostUserStore:false, VhostUserStorePath:"/var/run/kata-containers/vhost-user/", VhostUserStorePathList:[]string(nil), GuestHookPath:"", VMid:"", SELinuxProcessLabel:"", RxRateLimiterMaxRate:0x0, TxRateLimiterMaxRate:0x0, SGXEPCSize:0, EnableAnnotations:[]string(nil), GuestMemoryDumpPath:"", GuestMemoryDumpPaging:false}, NetmonConfig:virtcontainers.NetmonConfig{Path:"/tmp/kata-597178406/load-config-133133645/test/netmon", Debug:false, Enable:false}, AgentConfig:virtcontainers.KataAgentConfig{LongLiveConn:true, Debug:false, Trace:false, EnableDebugConsole:false, ContainerPipeSize:0x0, TraceMode:"", TraceType:"", KernelModules:[]string(nil)}, Console:"", InterNetworkModel:0, FactoryConfig:oci.FactoryConfig{Template:false, TemplatePath:"/run/vc/vm/template", VMCacheNumber:0x0, VMCacheEndpoint:"/var/run/kata-containers/cache.sock"}, Debug:false, Trace:false, JaegerEndpoint:"localhost", JaegerUser:"jaeger_user1", JaegerPassword:"jaeger_password1", DisableGuestSeccomp:false, DisableNewNetNs:false, SandboxCgroupOnly:false, Experimental:[]experimental.Feature(nil), EnablePprof:true}
                                actual  : oci.RuntimeConfig{HypervisorType:"qemu", HypervisorConfig:virtcontainers.HypervisorConfig{NumVCPUs:0x1, DefaultMaxVCPUs:0x4, MemorySize:0x800, DefaultBridges:0x1, Msize9p:0x2000, MemSlots:0xa, MemOffset:0x0, VirtioFSCacheSize:0x0, KernelParams:[]virtcontainers.Param{virtcontainers.Param{Key:"systemd.unit", Value:"kata-containers.target"}, virtcontainers.Param{Key:"systemd.mask", Value:"systemd-networkd.service"}, virtcontainers.Param{Key:"systemd.mask", Value:"systemd-networkd.socket"}, virtcontainers.Param{Key:"scsi_mod.scan", Value:"none"}, virtcontainers.Param{Key:"foo", Value:"bar"}, virtcontainers.Param{Key:"xyz", Value:""}}, HypervisorParams:[]virtcontainers.Param(nil), KernelPath:"/tmp/kata-597178406/load-config-133133645/test/kernel", ImagePath:"/tmp/kata-597178406/load-config-133133645/test/image", InitrdPath:"", FirmwarePath:"", MachineAccelerators:"", CPUFeatures:"", HypervisorPath:"/tmp/kata-597178406/load-config-133133645/test/hypervisor", HypervisorPathList:[]string(nil), HypervisorCtlPathList:[]string(nil), HypervisorCtlPath:"", JailerPath:"", JailerPathList:[]string(nil), BlockDeviceDriver:"virtio-scsi", HypervisorMachineType:"machineType", MemoryPath:"", DevicesStatePath:"", EntropySource:"/dev/urandom", SharedFS:"virtio-9p", VirtioFSDaemon:"/tmp/kata-597178406/load-config-133133645/test/virtiofsd", VirtioFSDaemonList:[]string(nil), VirtioFSCache:"none", VirtioFSExtraArgs:[]string(nil), FileBackedMemRootDir:"", FileBackedMemRootList:[]string(nil), PFlash:[]string{}, customAssets:map[types.AssetType]*types.Asset(nil), BlockDeviceCacheSet:false, BlockDeviceCacheDirect:false, BlockDeviceCacheNoflush:false, DisableBlockDeviceUse:true, EnableIOThreads:true, Debug:false, MemPrealloc:false, HugePages:false, VirtioMem:false, IOMMU:false, IOMMUPlatform:false, Realtime:false, Mlock:true, DisableNestingChecks:false, DisableImageNvdimm:false, HotplugVFIOOnRootBus:true, PCIeRootPort:0x2, BootToBeTemplate:false, BootFromTemplate:false, DisableVhostNet:false, EnableVhostUserStore:false, VhostUserStorePath:"/var/run/kata-containers/vhost-user/", VhostUserStorePathList:[]string(nil), GuestHookPath:"", VMid:"", SELinuxProcessLabel:"", RxRateLimiterMaxRate:0x0, TxRateLimiterMaxRate:0x0, SGXEPCSize:0, EnableAnnotations:[]string(nil), GuestMemoryDumpPath:"", GuestMemoryDumpPaging:false}, NetmonConfig:virtcontainers.NetmonConfig{Path:"/tmp/kata-597178406/load-config-133133645/test/netmon", Debug:false, Enable:false}, AgentConfig:virtcontainers.KataAgentConfig{LongLiveConn:true, Debug:false, Trace:false, EnableDebugConsole:false, ContainerPipeSize:0x0, TraceMode:"", TraceType:"", KernelModules:[]string(nil)}, Console:"", InterNetworkModel:0, FactoryConfig:oci.FactoryConfig{Template:false, TemplatePath:"/run/vc/vm/template", VMCacheNumber:0x0, VMCacheEndpoint:"/var/run/kata-containers/cache.sock"}, Debug:false, Trace:false, JaegerEndpoint:"localhost", JaegerUser:"jaeger_user1", JaegerPassword:"jaeger_password1", DisableGuestSeccomp:false, DisableNewNetNs:false, SandboxCgroupOnly:false, Experimental:[]experimental.Feature(nil), EnablePprof:true}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -62,3 +62,4 @@
                                   FileBackedMemRootList: ([]string) <nil>,
                                -  PFlash: ([]string) <nil>,
                                +  PFlash: ([]string) {
                                +  },
                                   customAssets: (map[types.AssetType]*types.Asset) <nil>,
                Test:           TestConfigLoadConfiguration
```

The error is about nil slice and empy slice. The question is why it happens. I thought the test has been run in CI.

My Go env is `go version go1.15.8 linux/amd64`.

Steps to reproduce:

1. clone the repo
2. cd src/runtime
3. make; make test

**Comment by jodh-intel** [comment]
Thanks for raising @zhsj. I wonder if this is a golang version specific issue? We use two versions documented here;

- https://github.com/kata-containers/kata-containers/blob/main/versions.yaml#L258

Could you or someone else test with golang 1.11.10 or 1.14.4?

**Comment by zhsj** [comment]
Fixed by 190f81342720581be0e66cfc60f9730806e49720
